accounts/src/inmemory_account_store.rs

```diff
@@ -165,33 +165,37 @@ impl AccountDataByCommitment {
                     }
                 }
             }
-        } else {
+        } else if commitment == Commitment::Finalized {
             // remove processed slot data
-            while self
-                .processed_accounts
-                .first_key_value()
-                .map(|(s, _)| *s)
-                .unwrap_or(u64::MAX)
-                <= slot
+            while self.processed_accounts.len() > 1
```

Comment:
not sure but it could be that there are processed blocks later than `slot` which got slashed. so they should also be removed

Reply:
No because processed accounts should atleast contain the last updated account state.